### PR TITLE
SALTO-6398: Add syncWorkspaceToFolder API

### PR DESCRIPTION
**Pull request by ori-moisis**
Add a new core API that syncs the elements from a workspace to a folder in an adapter format (SFDX) This also refactors the existing "calculatePatch" function which also deals with adapter formatted folders

---

_Additional context for reviewer_
No release notes yet since this is not an exposed command.
will add to the release notes once the CLI command is added

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/core/index.ts [hunk 1]
```diff
@@ -78,3 +78,4 @@ export {
 } from './src/local-workspace/remote_map'
 export { NoWorkspaceConfig } from './src/local-workspace/errors'
 export * from './src/types'
+export { calculatePatch, syncWorkspaceToFolder } from './src/core/adapter_format'
```
Comment (ori-moisis) [inline packages/core/index.ts:81]: `calculatePatch` was exported implicitly from this file by the `export * from ./src/api`

File: packages/core/src/core/adapter_format.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  return {
+    changes,
+    mergeErrors: [],
+    fetchErrors: [...(beforeLoadErrors ?? []), ...(afterLoadErrors ?? [])],
+    success: true,
+    updatedConfig: {},
+  }
+}
+
+type SyncWorkspaceToFolderArgs = {
```
Comment (dorsev) [inline packages/core/src/core/adapter_format.ts:164]: Super Nit\ or Question, should this be defined as `type SyncWorkspaceToFolderArgs = GetAdapterAndContextArgs & { baseDir: string } `?
 
that way, if GetAdapterAndContextArgs changes, this changes right away?

File: packages/core/src/core/adapter_format.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    resolvedElements.map(elem =>
+      toChange({
+        before: folderElementsByID[elem.elemID.getFullName()] as ChangeDataType,
+        after: elem as ChangeDataType,
+      }),
+    ),
+  )
+
+  log.debug(
+    'Loaded %d elements from folder %s, applying %d delete changes and %d addition changes to folder %s',
```
Comment (dorsev) [inline packages/core/src/core/adapter_format.ts:222]:
```suggestion
    'Loaded %d elements from folder %s, applying %d delete changes and %d addition(or modification) changes to folder %s',
``` 

just to make sure I understand this correctly.

File: packages/core/test/core/adapter_format.test.ts [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+  let mockAdapter: ReturnType<typeof createMockAdapter>
+
+  beforeEach(() => {
+    mockAdapter = createMockAdapter(mockAdapterName)
+    adapterCreators[mockAdapterName] = mockAdapter
+  })
+  afterEach(() => {
+    delete adapterCreators[mockAdapterName]
+  })
+  describe('when the folder has different elements from the workspace', () => {
```
Comment (dorsev) [inline packages/core/test/core/adapter_format.test.ts:222]: great looking test 🥇  was easy to read and helped to understand the behavior.

File: packages/core/src/core/adapter_format.ts [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+  accountName: string
+  baseDir: string
+  ignoreStateElemIdMapping?: boolean
+  ignoreStateElemIdMappingForSelectors?: ElementSelector[]
+}
+
+export type SyncWorkspaceToFolderResult = {
+  errors: ReadonlyArray<SaltoError>
+}
+export const syncWorkspaceToFolder = async ({
```
Comment (dorsev) [inline packages/core/src/core/adapter_format.ts:175]: Nit: Can we add timing to this function? so we can see how this behaves in large workspaces?

**Review by dorsev — APPROVED** [review]
Great job! LGTM.

none of my comments are blockers.

**Pull request merged**